rmb-sdk-go/peer/peer.go

```diff
@@ -260,6 +264,20 @@ func NewPeer(
 		return nil, err
 	}
 
+	// Hybrid approach: if RequireFunctionalRelayOnStartup is true, require at least one working relay at startup
+	if cfg.RequireFunctionalRelayOnStartup {
+		hasWorking := false
```

Comment:
you can replace the logic from 269 to 279 with IndexFun

```
	if cfg.RequireFunctionalRelayOnStartup {
		firstWorking := slices.IndexFunc(conns, func(c InnerConnection) bool {
			return c.TryConnect()
		})
		if firstWorking == -1 {

			....
        }
```

Reply:
This is a good use of `slices.IndexFunc`
Done!